### ISSUE TITLE
EOL

### ISSUE THREAD

**Issue by laktak** [hjson/hjson#84]
@dannyfritz
@dqsully
@iainhallam
@namusyaka
@trobro

I wanted to make you aware that I won't be maintaining this project any longer. See https://hjson.org

Thanks for your help!

**Comment by dqsully** [comment]
With your permission I would like to continue supporting this project. I think it would be a shame to let Hjson fade away, especially for how much work you put into it. My time is limited too, and I don't know nearly as many languages as you do, but I can still contribute my part.

**Comment by trobro** [comment]
I'm sorry to hear that, thanks for creating a great project!

**Comment by laktak** [comment]
@dqsully you are very welcome! I've changed the permission for all five of you to write. Let me know if/when you need anything else.

**Comment by dqsully** [comment]
Cool, sounds good. Thank you.

**Comment by dqsully** [comment]
Should we change the EOL notices to "Looking for maintainers?" I think you have enough influence with Hjson that the project could be entirely community-supported.

**Comment by sffc** [comment]
I evangelize Hjson through projects of mine, and I intend to continue doing so.  The existing Hjson ecosystem is strong; 1700 stars on GitHub is nothing to be ashamed of.  There is simply no other configuration file format that powerful, easy to for humans to read, and easy for computers to parse all at the same time.

I agree that this can and should be a community-driven project.  The message on hjson.org should reflect that.  I do not see any value in scaring potential users away by saying the project is "EOL".

**Comment by trobro** [comment]
I agree, I will continue to use and maintain hjson-cpp.

**Comment by laktak** [comment]
> Should we change the EOL notices to "Looking for maintainers?" I think you have enough influence with Hjson that the project could be entirely community-supported.

Please feel free to update the web site as you wish. I'm sorry that the notice was a bit premature and I'd be very happy if you can get the project community supported.

If you like to continue to use my 'publishing system' I've added a build script to the pages folder. You only need to update the markdown files to generate the html for github pages. Let me know if you get stuck.

Also I noticed that Github has a new teams feature. I'm not familiar with it but I can activate it for you if you wish.

I'd be happy to assist with any issues, I just don't have the time or motivation to update, test & publish the implementations that I have handled before.

**Comment by dqsully** [comment]
I've made a PR (#85) to update the homepage. If any of you have some feedback I'd be glad to take it.

**Comment by sffc** [comment]
I'm happy to help maintain hjson-py and hjson-ruby.

**Comment by NuSkooler** [comment]
I'm watching & can help with JS where needed.

**Comment by jonbodner** [comment]
@mspiegel and I used hjson-go in github.com/capitalone/checks-out . We'd be interested in maintaining the Go library, if you don't have anyone else lined up.

**Comment by laktak** [comment]
@dqsully I've given you admin on all repositories - this should allow you to manage the collaborators.

Not sure how we can handle the publishing. The npm packages were on my account before we created the hjson org. Can travis publish to npm without exposing the keys?

**Comment by dqsully** [comment]
Yes, Travis CI can [handle it](https://docs.travis-ci.com/user/encryption-keys/). Travis can encrypt and hide the keys, but if someone approves a PR with bad code then the NPM keys could leak (see what happened to [ESLint](https://gist.github.com/hzoo/51cb84afdc50b14bffa6c6dc49826b3e)).

**Comment by stardust85** [comment]
Hi, today I found that JSON5 exists and it looks like it tries to solve very similar issues and hjson. https://github.com/json5/json5 It has more stars than hjson. Let's move our effort to json5!

**Comment by dqsully** [comment]
While JSON5 also improves upon JSON, Hjson is more focused on being a human-readable configuration language. Also, JSON5 doesn't support nearly as many languages/platforms as Hjson does. And even if it did, I wouldn't want to abandon everyone who currently uses Hjson.

**Comment by laktak** [comment]
My suggestion to the community is to come up with a new name. 

"Hjson" is problematic because it has JSON in the name. Many people dismiss it outright because they think of it as an alternative or replacement for JSON, which it is not. It would be way easier to explain what it actually is without that assumption in their heads.

**Comment by NuSkooler** [comment]
JSON5 looks interesting -- but I agree, HJSON is a different beast. What would be great is if efforts could be combined in compatible ways.

**Comment by sffc** [comment]
I've read about JSON5.  Hjson is generally designed for a different purpose, config files rather than a data transfer format.  Here are some examples:

1. Unquoted strings (!)
2. Better multiline strings
3. Optional trailing commas

In my mind the decision is Hjson vs. YAML, not Hjson vs. JSON5.  (I prefer Hjson over all three.)

If we rebrand Hjson to a different name, we should do it at a time when we're prepared for a big PR push: change all the packages at the same time, get a new thread on HackerNews, etc.

**Comment by trobro** [comment]
I also prefer Hjson, but I'm not overly concerned about the popularity of the project. I maintain hjson-cpp because I use it myself. If anyone else wants to use it too then that's a bonus. The name Hjson worked well on me, because I wanted to use something very simple like JSON but with comments, not something complex like YAML.

**Comment by stardust85** [comment]
I looked at json5 closer and I realized that hjson is really more convenient for writing. You don't need to write commas at the end of every line, you don't need to qoute neither keys, nor values! I really start to love hjson! :)

**Comment by princec** [comment]
hjson really is far superior for its use case. I would suggest ignoring json5 and letting it go its own path.

**Comment by 3F** [comment]
Sadly to hear about abandoned status :( just found you.

This is really great alternative even for YAML that I still don't like (compared to an classic JSON model). 
Today I'm also watching JSON5 project with more stars than HJSON, but there is something wrong -_-

HJSON easily solves most of the JSON inconveniences and it still classically structured as JSON. I like it!

I think, I will consider HJSON anyway for my future use even it's discontinued now.

I also like how an multiline strings implementation will help to avoid some nightmares, for example:
https://github.com/3F/Conari/blob/master/.vssbe#L353

```json

"Mode": {
	"$type": "net.r_eg.vsSBE.Events.ModeScript, vsSolutionBuildEvent",
	"Type": "Script",
	"Command": "#[7z pack.files({ \n        \"$(odir)\\\\lib\\\\net$(netstamp)\\\\*.*\",\n        \"$(odir)\\\\*.*\"\n    }, \n    \"$(dbin)\\\\Conari_v$(pVer).zip\", \n    { \"$(odir)\\\\*.nuspec\" })]",
...
```

In past (~2014) the `Command` property above was extracted to an array of strings, 
to be more understandable for a just quick human view from config file.

it was like (JSON):

```yml
"Command__": [
  "#[7z pack.files({ ",
  "        \"$(odir)\\\\lib\\\\net$(netstamp)\\\\*.*\",",
  "        \"$(odir)\\\\*.*\"",
  "    }, ",
  "    \"$(dbin)\\\\Conari_v$(pVer).zip\", ",
  "    { \"$(odir)\\\\*.nuspec\" })]"
]
```

And I really like how HJSON may help with it:

```yml
Mode:
{
  $type: net.r_eg.vsSBE.Events.ModeScript, vsSolutionBuildEvent
  Type: Script
  Command:
    '''
    #[7z pack.files
    (
        {
            "$(odir)\\lib\\net$(netstamp)\\*.*",
            "$(odir)\\*.*"
        }, 
        "$(dbin)\\Conari_v$(pVer).zip", 
        { 
            "$(odir)\\*.nuspec" 
        }
    )]
    '''
```

comments, quote marks, trailing commas, ...

Thus, Guys! This is really great and important superset of JSON just to be with us.

**Comment by trobro** [comment]
> Sadly to hear about abandoned status :( just found you.

The rumors of abandonment are greatly exaggerated. ;)  The creator has left, but the project is still being maintained by me and others, since we agree with you that Hjson is a really nice format that deserves to live on.

**Comment by sffc** [comment]
ICU 64 (as in [International Components for Unicode](https://github.com/unicode-org/icu/), the Unicode Consortium's widely used low-level i18n library) is rolling out [a new configuration file](https://github.com/unicode-org/icu/blob/5630114b7add634f13eca0ca50fcf3bfc7c694fd/docs/userguide/data_buildtool.md#icu-data-configuration-file) that supports Hjson this month.

**Comment by antonmedv** [comment]
Hello 👋🏻

I'm starting a new spec which will be very close to Hjson: https://maml.dev

```py
{
  project: "MAML"
  tags: [
    "minimal"
    "readable"
  ]

  # Comments
  notes: """
This is a multiline strings.
Keeps formatting as-is.
"""
}
``` 

Please, take a look if you're interested.
